File: 1-TwoSum/1-TwoSum.py
# Last updated: 02/08/2025, 19:00:09
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        arr = []
        f=0
        for i in range(0, len(nums)):
            for j in range(0, len(nums)):
                if (nums[i]+nums[j]==target) and i != j:
                    arr.append(i)
                    arr.append(j)
                    f=1
            if f==1:
                break     
        return arr
        

File: 1-TwoSum/1-TwoSum.java
// Last updated: 19/12/2025, 10:22:38
1class Solution {
2    public int[] twoSum(int[] nums, int target) {
3        HashMap<Integer, Integer> map = new HashMap<>();
4        int[] ans = new int[2];
5        for(int i=0; i<nums.length; i++) {
6            int compliment = target - nums[i];
7            if(map.containsKey(compliment)) {
8                ans[0] = map.get(compliment);
9                ans[1] = i;
10                break;
11            }
12            map.put(nums[i], i);
13        }
14        return ans;
15    }
16}